### PR TITLE
Rename "tests" to "utils" + remove deprecated parameters

### PR DESCRIPTION
**Pull request by pcercuei**
Move the whole "tests" folder to now be called "utils".

Remove the "-n", "-x" and "-s" parameters of those utility programs.

### CODE REVIEW

File: README_BUILD.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ Cmake Options          | Default | Target | Description
 'COMPILE_WARNING_AS_ERROR' | OFF |    All | Make all C warnings into errors     |
 `CPP_BINDINGS`         | OFF |        All | Install C++ bindings (C++17 required for examples) |
 `PYTHON_BINDINGS`      | OFF |        All | Install PYTHON bindings                            |
-`WITH_TESTS`           |  ON |        All | Build the test programs (iio-utils)                |
+`WITH_UTILS`           |  ON |        All | Build the utilitary programs (iio-utils)           |
```
Comment (tfcollins) [inline README_BUILD.md:47]: Should be utility not utilitary

**Review by tfcollins — CHANGES_REQUESTED** [review]
Remove the doc for the x case as well https://github.com/analogdevicesinc/libiio/blob/f4e6df15f86c49ed265db47f471252f35d215d7e/utils/iio_common.c#L195

**Comment by rgetz** [comment]
https://github.com/analogdevicesinc/libiio/blob/main/tests/iio_common.h#L50 needs update as well.

**Comment by rgetz** [comment]
looks good to me. Thanks

**Review by tfcollins — APPROVED** [review]
I think you got everything. Merge please :)

**Pull request merged**